lib/request.js

```diff
@@ -1,192 +1,159 @@
+      throw new Error('hostname is a required option');
+    }
   }
 
-  this.options.port = this.options.port || 80;
-  this.options.ssl = this.options.ssl || false;
+  oAuthConfig() {
+    return oAuth({
+      consumer: {
+        public: this.options.consumerKey,
+        secret: this.options.secret
+      },
+      'signature_method': 'HMAC-SHA256',
+      version: null,
+      'last_ampersand': false
+    });
+  }
 
-  // Set the log level to errors only by default
-  logger.level = this.options.logLevel || 0;
-};
+  requestData(path, method) {
+    return {
+      url: this.options.hostname +
+        path,
+      method: method,
+      data: {}
+    };
+  }
 
-Request.prototype.setHeaders = function(options) {
-  if (this.options.headers) {
-    for (var key in this.options.headers) {
-      options.headers[key] = this.options.headers[key];
+  formatQuery(authorize) {
+    let query = '?';
+    for (let key in authorize) {
+      const val = key === 'oauth_signature' ?
+        querystring.escape(authorize[key]) : authorize[key];
+      query += key + '=' + val + '&';
     }
+    return query.substr(0, query.length - 1);
   }
-  return options;
-};
-
-Request.prototype.setBasicAuth = function(options) {
-  options.auth = this.options.consumerKey +
-    ':' + this.options.secret;
-  return options;
-};
-
-Request.prototype.oAuthConfig = function() {
-  return OAuth({
-    consumer: {
-      public: this.options.consumerKey,
-      secret: this.options.secret
-    },
-    'signature_method': 'HMAC-SHA256',
-    version: null,
-    'last_ampersand': false
-  });
-};
-
-Request.prototype.requestData = function(path, method) {
-  return {
-    url: 'http://' +
-      this.options.hostname +
-      path,
-    method: method,
-    data: {}
-  };
-};
-
-Request.prototype.formatQuery = function(authorize) {
-  var query = '?';
-  for (var key in authorize) {
-    var val = key === 'oauth_signature' ?
-      querystring.escape(authorize[key]) : authorize[key];
-    query += key + '=' + val + '&';
-  }
-  return query.substr(0, query.length - 1);
-};
-
-Request.prototype.addOAuth = function(options) {
-  var oauth = this.oAuthConfig();
-  var request = this.requestData(options.path, options.method);
-  var authorize = oauth.authorize(request);
-  options.path += this.formatQuery(authorize);
-  return options;
-};
-
-Request.prototype.completeRequest = function(method, path, data, cb) {
-  var _this = this;
-
-  logger.info('Requesting Data from: https://' +
-    this.options.hostname + path + ' Using the ' +
-    method + ' method');
-
-  var protocol = this.options.ssl ? require('https') : require('http');
-  var dataString = JSON.stringify(data);
-  var options = {
-    hostname: this.options.hostname,
-    path: path,
-    method: method.toUpperCase(),
-    port: this.options.port,
-    headers: {
-      'User-Agent': 'node-woocommerce/1.0.0',
-      'Content-Type': 'application/json',
-      'Accept': 'application/json'
```

Comment:
They don't work on all servers, I tried it on a guy that's having problems with the existing one, and it returns an error saying it doesn't understand the request.